src/rules/no_import_prefix.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+use super::{Context, LintRule};
+use crate::handler::{Handler, Traverse};
+use crate::Program;
+use deno_ast::view::{CallExpr, Callee, Expr, ImportDecl, Lit};
+use deno_ast::SourceRanged;
+
+#[derive(Debug)]
+pub struct NoImportPrefix;
+
+const CODE: &str = "no-import-prefix";
+const MESSAGE: &str =
+  "Inline 'npm:*', 'jsr:*' or 'http:*' dependency not allowed";
+const HINT: &str = "Add it to the 'imports' section in 'deno.json' instead";
+
+impl LintRule for NoImportPrefix {
+  fn tags(&self) -> &'static [&'static str] {
+    &[]
+  }
```

Comment:
I'm worrying that ppl will not use this rule because it's not enabled by default. Let's think about a way we could conditionally enable it in `deno` if we discover `deno.json` or `package.json` - in such cases you're not writing a one-off script but have a project and you should centralize dependencies.